### DISCUSSION TITLE
'Used by' spam abusing `go.mod`

### DISCUSSION THREAD

**Discussion by rhysd** [community/community#20984] [Other Feature Feedback, Questions, & Ideas] [answered]
Some users are abusing `go.mod` to display their repositories in 'Used by' section of famous repositories.

Actual example: https://github.com/anas-elgarhy/anas-elgarhy/blob/d1981ebb7968cd7dc43ae38960d932c87ba1e9e6/go.mod

```
module github.com/Anas-Elgarhy

go 1.18

require (
	github.com/Anas-Elgarhy/JCmusIntegrationWithDiscord v1.1.0 // indirect
	github.com/Anas-Elgarhy/dwm v0.0.0-20220519083658-f84ab69cee14 // indirect
	github.com/Anas-Elgarhy/slstatus v0.0.0-20220501211816-a3fb8788206d // indirect
	github.com/alacritty/alacritty v0.10.1 // indirect
	github.com/cmus/cmus v2.9.1+incompatible // indirect
	github.com/drkhsh/slstatus v0.0.0-20201130203214-84a2f117a32f // indirect
	github.com/garbeam/dwm v0.0.0-20190202125042-cb3f58ad0699 // indirect
	github.com/git/git v2.36.1+incompatible // indirect
	github.com/golang/go v0.0.0-20220519071347-128279e5034c // indirect
	github.com/jetBrains/kotlin v1.6.21 // indirect
	github.com/neovim/neovim v0.7.0 // indirect
	github.com/rust-lang/rust v0.0.0-20220519040440-50872bdb99c9 // indirect
	github.com/tmux/tmux v0.0.0-20220514191646-ead75c2d512c // indirect
	github.com/zsh-users/zsh v0.0.0-20220520215050-a54d70457934 // indirect
)
```

In [neovim/neovim](https://github.com/neovim/neovim) repository:

<img width="272" alt="" src="https://user-images.githubusercontent.com/823277/180114054-d56c2ea6-c127-492f-8458-b7624dbe8a93.png">

The `go.mod` file is put only for spam so the repository doesn't actually use them.

How can we prevent such spams?

**Comment by kecci** [comment]
I'm sure your dependencies also have another dependencies, so it's look like library call another library until the root.

**Comment by gpanders** [comment]
I'd like to bump this post and request a response from GitHub staff. This user in particular abuses the dependency graph feature to put their name on many public repos, and the maintainers of these repos have no ability to remove it.

**Reply by gpanders** [reply]
Currently listed on

- [tmux/tmux](https://github.com/tmux/tmux)
- [neovim/neovim](https://github.com/neovim/neovim)
- [zsh-users/zsh](https://github.com/zsh-users/zsh)
- [git/git](https://github.com/git/git)
- [cmus/cmus](https://github.com/cmus/cmus)

**Comment by gpanders** [comment]
Also present on https://github.com/smolck/uivonim. Users should not be able to add arbitrary text to a repo without the consent (or even the inability to remove it!) of the repo owner.

**Comment by queenofcorgis** [comment, marked as answer]
Hi all 👋🏼  We really appreciate you flagging this. The best route to get this to the proper GitHub team is to use our abuse reporting tools. Here's all the info:

- [Reporting abuse or spam](https://docs.github.com/en/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam)
- [Reporting a user](https://docs.github.com/en/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam#reporting-a-user)
- [Reporting an issue or pull request](https://docs.github.com/en/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam#reporting-an-issue-or-pull-request)
- [Reporting a comment](https://docs.github.com/en/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam#reporting-a-comment)

You can report behavior and content that violates community guidelines and terms. I am going to lock this post, but for this and any future incidents, please refer to those links.

Thank you!
